vault/identity_store_util.go

```diff
@@ -65,10 +67,15 @@ func (i *IdentityStore) loadArtifacts(ctx context.Context, isActive bool) error
 				persistMerges = true
 			}
 
-			// Since we're reloading entities, we need to inform the ConflictResolver about it so that it can
-			// clean up data related to the previous load.
+			// Since we're reloading entities, we need to inform the ConflictResolver
+			// about it so that it can clean up data related to the previous load.
 			i.conflictResolver.Reload(ctx)
 
+			// We also need to start again from empty state or wild things can happen!
+			if err := i.resetDB(); err != nil {
+				return err
+			}
```

Comment:
This is the critical fix that avoids double-renaming (i.e. appending UUIDs twice) for duplicates.

Reply:
I actually think this might be incorrect. If we make it through with no duplicates and reload to persist the second time, this will effectively wipe out ALL memDB state and only re-do the loadEntities work. We might need to rethink  the approach or come up with a different solution!